src/utilities/collection-utils.test.ts

```diff
@@ -442,5 +476,61 @@ describe("CollectionUtils", () => {
             );
             expect(equals).toBe(true);
         });
+
+        it("When source array has no values it, then returns the source array", () => {
+            // Arrange
+            const arr = [];
+
+            // Act
+            const result = CollectionUtils.replaceElementAt(arr, 1, "replaced");
+
+            // Assert
+            expect(result).toBe(arr);
+        })
+
+        it("When source array has 1 item it, then returns new array with 1 item", () => {
+            // Arrange
+            const arr = ["test"];
+            const expected = ["replaced-test"];
+
+            // Act
+            const result = CollectionUtils.replaceElementAt(arr, 2, "replaced-test");
+
+            // Assert
+            const equals = CollectionUtils.equalsBy(
+                (s: string) => s,
+                result,
+                expected
+            );
+            expect(equals).toBe(true);
+        })
+
+        it("When index is last element of source it, then returns new array with value at the end", () => {
+            // Arrange
+            const arr = ["zero", "one", "two", "three", "four"];
+            const expected = ["zero", "one", "two", "three", "replaced"];
+
+            // Act
+            const result = CollectionUtils.replaceElementAt(arr, 4, "replaced");
+
+            // Assert
+            const equals = CollectionUtils.equalsBy(
+                (s: string) => s,
+                result,
+                expected
+            );
+            expect(equals).toBe(true);
```

Comment:
Had to double check what method was under test here - adding the region around each method's respective `describe()` block would help 👍